test/node/advanced.spec.js

```diff
@@ -47,44 +75,99 @@ describe('advanced', () => {
   describe('React', () => {
     describe('element (with circular references)', () => {
       it('compares without warning or errors', () => {
-        const testRenderer = ReactTestRenderer.create(React.createElement(Container));
-        testRenderer.update(React.createElement(Container));
+        const reactApp = ReactTestRenderer.create(
+          React.createElement(ReactContainer)
+        );
+        reactApp.update(React.createElement(ReactContainer));
+        assert.strictEqual(warnStub.callCount, 0);
+      });
+      it('elements of same type and props are equal', () => {
+        const reactApp = ReactTestRenderer.create(
+          React.createElement(ReactContainer)
+        );
+        reactApp.update(React.createElement(ReactContainer));
+        assert.strictEqual(reactChildRenderSpy.callCount, 1);
+      });
+      it('elements of same type with different props are not equal', () => {
+        const reactApp = ReactTestRenderer.create(
+          React.createElement(ReactContainer)
+        );
+        reactApp.update(React.createElement(ReactContainer, { title: 'New' }));
+        assert.strictEqual(reactChildRenderSpy.callCount, 2);
+      });
+    });
+  });
+
+  describe('Preact', () => {
+    let cleanupJsDom;
+
+    before(() => {
+      cleanupJsDom = jsdom();
+    });
+
+    after(() => {
+      PreactTestRenderer.cleanup();
+      cleanupJsDom();
+    });
+
+    describe('element (with circular references)', () => {
+      it('compares without warning or errors', () => {
+        PreactTestRenderer.render(
+          Preact.createElement(PreactContainer)
+        );
+        PreactTestRenderer.render(
+          Preact.createElement(PreactContainer)
+        );
         assert.strictEqual(warnStub.callCount, 0);
       });
       it('elements of same type and props are equal', () => {
-        const testRenderer = ReactTestRenderer.create(React.createElement(Container));
-        testRenderer.update(React.createElement(Container));
-        assert.strictEqual(childRenderSpy.callCount, 1);
+        const { container } = PreactTestRenderer.render(
+          Preact.createElement(PreactContainer)
+        );
+        Preact.render(PreactContainer, container);
+        assert.strictEqual(preactChildRenderSpy.callCount, 1);
       });
       it('elements of same type with different props are not equal', () => {
-        const testRenderer = ReactTestRenderer.create(React.createElement(Container));
-        testRenderer.update(React.createElement(Container, { title: 'New' }));
-        assert.strictEqual(childRenderSpy.callCount, 2);
+        const { container } = PreactTestRenderer.render(
+          Preact.createElement(PreactContainer)
+        );
+        Preact.render(PreactContainer, container);
```

Comment:
The offending line - we need to supply new props to update this component

Reply:
isn't this because we aren't calling `createElement` on it and just passing in a class? This isn't properly converted to a `vnode`

Reply:
I only want to trigger a re-render (similar to React's `.update()`) of this component, does `createElement` handle updates if the element is already known to exist?

Reply:
Oh okay, I misunderstood the problem.

```
        const { container, rerender } = PreactTestRenderer.render(
          Preact.createElement(PreactContainer)
        );
        rerender(Preact.createElement(PreactContainer));
```

Should do the trick